src/main/scala/shoppinglist/domain/ShoppingList.scala

```diff
@@ -0,0 +1,66 @@
+package shoppinglist.domain
+
+import shoppinglist.api.{ListId, ListItemId, OwnerId, ShoppingListManagementError}
+import ShoppingListManagementError._
+import zio.{UIO, ZIO}
+
+final case class ShoppingList(
+    id: ListId,
+    ownerId: OwnerId,
+    name: String,
+    items: List[ListItem]
```

Comment:
I like  when the model is descriptive and reflect all operations you could do with it. 
But in real world items list could be huge, and it won't be effective from language perspective.

Reply:
You're right. It's all trade-offs.